### DISCUSSION TITLE
How to reduce memory usage

### DISCUSSION THREAD

**Discussion by pyking-wang** [AvaloniaUI/Avalonia#12141] [Q&A]
My Avalonia project always consumes a lot of computer memory. Is there any way I can reduce it? I hope to receive some suggestions, whether it's programming or structure. Thank you.
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/090e2a26-8d91-4e65-9eed-efe2cbcd957f)

**Comment by timunie** [comment]
do you have some high-res images? If yes reduce their size by using `Bitmap.DecodeToWidth()`

**Reply by pyking-wang** [reply]
I don't have many high-resolution images, but I have many of these icons.
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/16c58c0c-4c99-4cc4-8641-835f874f6695)
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/51092c82-9cf8-4888-858a-f6ba08b8e705)
Will these affect or what better way to handle them？

**Reply by timunie** [reply]
could be possible, yes. Just check pixel size (not image file size) as each pixel is uncompressed in memory.

**Comment by rabbitism** [comment]
my suggestion is to check the released App instead of checking visual studio memory consumption

**Reply by pyking-wang** [reply]
I think the Avalonia project should not allow visual studio to consume so much memory during the development process. I will pay attention to thereleased App later.

**Reply by Tamnac** [reply]
How would the project control that? It's just a bunch of files. Visual Studio is the actual running process here
